apps/performance/default-onpush/src/app/person-list.component.ts

```diff
@@ -1,68 +1,42 @@
-import { Component, Input } from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  OnInit,
+} from '@angular/core';
 
-import { CDFlashingDirective } from '@angular-challenges/shared/directives';
 import { CommonModule } from '@angular/common';
 import { FormsModule } from '@angular/forms';
-import { MatChipsModule } from '@angular/material/chips';
-import { MatFormFieldModule } from '@angular/material/form-field';
-import { MatInputModule } from '@angular/material/input';
-import { MatListModule } from '@angular/material/list';
+import { PersonInputComponent } from './input.component';
+import { ListComponent } from './list.component';
 
 @Component({
   selector: 'app-person-list',
   standalone: true,
-  imports: [
-    CommonModule,
-    FormsModule,
-    MatListModule,
-    MatFormFieldModule,
-    MatInputModule,
-    MatChipsModule,
-    CDFlashingDirective,
-  ],
+  imports: [CommonModule, FormsModule, PersonInputComponent, ListComponent],
+  changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
     <h1 cd-flash class="text-center font-semibold" title="Title">
       {{ title | titlecase }}
     </h1>
 
-    <mat-form-field class="w-4/5" cd-flash>
-      <input
-        placeholder="Add one member to the list"
-        matInput
-        type="text"
-        [(ngModel)]="label"
-        (keydown)="handleKey($event)" />
-    </mat-form-field>
-
-    <mat-list class="flex w-full">
-      <div *ngIf="names?.length === 0" class="empty-list-label">Empty list</div>
-      <mat-list-item
-        *ngFor="let name of names"
-        cd-flash
-        class="text-orange-500">
-        <div MatListItemLine class="flex justify-between">
-          <h3 title="Name">
-            {{ name }}
-          </h3>
-        </div>
-      </mat-list-item>
-      <mat-divider *ngIf="names?.length !== 0"></mat-divider>
-    </mat-list>
+    <app-person-input (personName)="getPersonName($event)"></app-person-input>
+    <app-list-component [names]="personNames"></app-list-component>
   `,
   host: {
     class: 'w-full flex flex-col items-center',
   },
 })
-export class PersonListComponent {
+export class PersonListComponent implements OnInit {
   @Input() names: string[] = [];
   @Input() title = '';
-
-  label = '';
-
-  handleKey(event: any) {
-    if (event.keyCode === 13) {
-      this.names?.unshift(this.label);
-      this.label = '';
-    }
+  personNames: string[] = [];
+  ngOnInit(): void {
+    this.personNames = this.names;
+  }
+  getPersonName(event: string) {
+    this.personNames = [];
+    this.names?.unshift(event);
+    this.personNames = [...this.names]; // not sure this is the best way. can I use changeDetectoRef instead of this ?
```

Comment:
you should be declarative and you can now use signal inputs 👍

Reply:
Funny enough - a signal implementation can look very similar.   Now that signals are immutable and the spread operator can be used to update a signal.  See this [YouTube video](https://www.youtube.com/watch?v=QOiWLGV9Yrg).